go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -78,7 +78,28 @@ func (tm *TabletManager) RestoreData(ctx context.Context, logger logutil.Logger,
 	if tm.Cnf == nil {
 		return fmt.Errorf("cannot perform restore without my.cnf, please restart vttablet with a my.cnf file specified")
 	}
-	return tm.restoreDataLocked(ctx, logger, waitForBackupInterval, deleteBeforeRestore)
+	// Tell Orchestrator we're stopped on purpose for some Vitess task.
+	// Do this in the background, as it's best-effort.
+	go func() {
+		if tm.orc == nil {
+			return
+		}
+		if err := tm.orc.BeginMaintenance(tm.Tablet(), "vttablet has been told to Restore"); err != nil {
```

Comment:
from our experiment, `tm.Tablet()` seems to give an out dated state and we are getting `cannot find mysql port error`, we ended up using `tm.tmState.Tablet()`. Let me know if you also experience it.

Reply:
`tm.Tablet()` simply returns `tm.tmState.Tablet()` so they should be equivalent. I do see that it is possible for a race condition to occur between `checkMysql` and `handleRestore`.